### PR TITLE
Weight Initialization & Layer-Specific Learning Rates

### PR DESCRIPTION
**Pull request by devyhia**
@amoudgl @sydney0zq After training some models on Caffe and checking the weights there, the only differences I found were:
1. Biases are initialized to 1 in Caffe (was not the case in PyTorch).
2. Weights are normally initialized with mean=0 and std= 0.005
3. Learning rates for weights and biases are different from the "overall learning rate". PyTorch does not provide support for lr_mult feature (in a direct way I mean).

The following update should show you a direct improvement on your training. For me, I get the same loss from step 100, that I used to get after ~80000 steps or so. Now, I get loss ~= 150 in the first 1000 steps which is very similar/comparable to the caffe model. Note, everything else is fixed. I am using the same training/batching code for both models.

### CODE REVIEW

**Comment by theodoruszq** [comment]
Good job. Let’s see the final results. Maybe it will cost 2 days.

**Comment by amoudgl** [comment]
> Now, I get loss ~= 150 in the first 1000 steps which is very similar/comparable to the caffe model.

Are you getting loss of ~150 on a batch of 11 samples? If I remember correctly, the current implementation also yields similar loss after 100 steps or so on a batch of 11 samples.

> I am using the same training/batching code for both models.

What's your batching technique? Are you using the current implementation for forming batches or you implemented the one there in the author's GOTURN implementation?

**Comment by devyhia** [comment]
> Are you getting loss of ~150 on a batch of 11 samples? If I remember correctly, the current implementation also yields similar loss after 100 steps or so on a batch of 11 samples.

Nope. On a batch of 50 images.

> What's your batching technique? Are you using the current implementation for forming batches or you implemented the one there in the author's GOTURN implementation?

Nope, I used the following code: https://github.com/devyhia/PY-GOTURN/commit/0a18a17ce68cead3237df3284a0bba6d87959672

You could check this branch out: https://github.com/devyhia/PY-GOTURN/tree/pytorch-train

**Comment by amoudgl** [comment]
Thanks a lot, @devyhia for this PR! I implemented the batch formation procedure of GOTURN in pytorch (#13) and I am able to reproduce the loss of ~150 after 1000 iterations as you stated. This patch indeed improves the training performance.
